python/lsst/obs/base/butler_tests.py

```diff
@@ -0,0 +1,186 @@
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import abc
+import inspect
+import unittest
+import collections
+
+
+class ButlerGetTests(with_metaclass(abc.ABCMeta)):
+    """
+    Tests of obs_* Butler get() functionality.
+
+    In the subclasses's setUp():
+        * Instantiate a Butler on the testdata repository as self.butler.
+        * Set self.dataIds to a dictionary of exposure names:dataId of that
+          exposure in the testdata repository, with unittest.SkipTest as the
+          value for any exposures you do not have/do not want to test. It must
+          contain a valid 'raw' dataId, in addition to 'bias','flat','dark',
+          which may be set to SkipTest. For example::
+              self.dataIds = {'raw': {'visit': 1, 'filter': 'g'},
+                              'bias': {'visit': 1},
+                              'flat': {'visit': 1},
+                              'dark': unittest.SkipTest
+        * Call setUp_butlerGet() to fill in required parameters.
+    """
+
+    def setUp_butlerGet(self,
+                        ccdExposureId_bits=None,
+                        exposureIds=None,
+                        filters=None,
+                        exptimes=None,
+                        detectorIds=None,
+                        detector_names=None,
+                        detector_serials=None,
+                        dimensions=None,
+                        sky_origin=None,
+                        raw_subsets=None,
+                        good_ccds=None,
+                        bad_ccds=None,
+                        linearizer=None
+                        ):
+        """
+        Set up the necessary variables for butlerGet tests.
+
+        ccdExposureId_bits: int
+            expected value of ccdExposureId_bits
+        exposureIds: dict
+            dict of exposure: exposure_id
+        filters: dict
+            dict of exposure: filter name
+        exptimes: dict
+            dict of exposure: exposure time
+        detector_names: dict
+            dict of exposure: detector name
+        detectorIds: dict
+            dict of exposure: detectorId
+        detector_serials: dict
+            dict of exposure: detector serial
+        dimensions: dict
+            dict of exposure: dimensions (as a geom.Extent2I)
+        sky_origin: tuple of float
+            Longitude, Latitude of 'raw' exposure
+        raw_subsets: tuple of (kwargs, int)
+            keyword args and expected number of subsets for butler.subset('raw', **kwargs)
+        good_ccds: list of int
+            list of valid ccd numbers
+        bad_ccds: list of int
+            list of invalid ccd numbers
+        linearizer: dict
+            'type':LinearizerType, and the converted values for 1,1000,30,000, for dataId['raw'],
+            or unittest.SkipTest to not test the linearizer.
```

Comment:
As with the dataId skipping, I want to make skipping the test be explicit, instead of implicit.